### DISCUSSION TITLE
In data viewer, prevent the Refresh, Hide Filters & Clear All Filters buttons from disappearing when scrolling horizontally

### DISCUSSION THREAD

**Discussion by charris-msft** [microsoft/vscode-jupyter#9761] [New Ideas]
I work with some fairly wide data frames - 40-70 columns, and since I'm no expert, I end up tweaking my code and refreshing the data viewer a lot! This is no problem when I'm primarily working with the first 10 or so columns, but when I'm scrolled to the right side, I have to scroll all the way back to the left in order to see the refresh button.

- Yes, I know that `Home` & `ctrl+Home` will take me back to the left, but when I'm exploring with the mouse, I don't want to switch to the keyboard.
- I would use a hotkey combo to refresh, but there doesn't seem to be one, and that's not intuitive for everyone.

Here is my recommendation:
![image](https://user-images.githubusercontent.com/74415662/165123817-3688125c-6bbf-4bdc-be53-f933e53f514f.png)

**Comment by rchiodo** [comment]
Actually there's a keyboard shortcut you can use. 

![image](https://user-images.githubusercontent.com/19672699/165628781-69489b9d-ce8a-4e50-906a-d5627ac6447a.png)

That should refresh the data viewer.

**Reply by charris-msft** [reply]
Ahhh! I should have looked there!

I do still think it would be beneficial to move the buttons, because the screen is much more discoverable than the keyboard shortcut menu. But I would also love to see the tool tips show the keyboard shortcuts (where available) too.

**Reply by rchiodo** [reply]
Thanks. I created two new issues for your suggestions.

**Reply by charris-msft** [reply]
Thanks! 

While I'm talking to Santa it would also be super helpful to have some kind of activity indicator (spin the refresh button's icon?) to acknowledge when I press the Ctrl+R shortcut to refresh the data viewer.

**Reply by rchiodo** [reply]
There should already be a blue progress bar on the top (well if takes more than like half a second). Yeah if I load a big enough data frame and hit CTRL+R, it will put up a progress bar along the top of the data viewer window.

**Reply by rchiodo** [reply]
Maybe you can't see it if you're scrolled.

**Reply by rchiodo** [reply]
No even scrolled it shows up.

**Reply by charris-msft** [reply]
You are absolutely right!
I believe my issue is that the time from when I hit ctrl+R to when the refresh starts - or at least the progress bar starts - can take 5 or more seconds and I'm definitely distracted by something else within that time. I think it's taking so long to start because I'm running at the limits of my 32GB of memory.

Now that I waited for it, I realized that I have seen it before. Most of my dataframes tend to be either very small samples, which don't result in a progress bar, or quite large ones (100K-10M rows x ~50 columns).

Thanks - I'm happy to know that it's there already! sorry for the distraction!
